internal/services/resource/resource_group_resource.go

```diff
@@ -195,6 +195,9 @@ func resourceResourceGroupDelete(d *pluginsdk.ResourceData, meta interface{}) er
 		err = pluginsdk.Retry(10*time.Minute, func() *pluginsdk.RetryError {
 			results, err := resourceClient.ListByResourceGroupComplete(ctx, id.ResourceGroup, "", "provisioningState", utils.Int32(500))
 			if err != nil {
+				if strings.Contains(err.Error(), "could not be found") {
```

Comment:
Error messages can change without us knowing, I wonder if we can check the response code of `results` using `response.WasNotFound(results.HttpResponse)`  instead?